Modules/CalcDefence-3_0.lua

```diff
@@ -685,4 +738,124 @@ function calcs.defence(env, actor)
+	output.SecondMinimumSpellProjectileDamageChance = m_second_min({output.PhysicalSpellProjectileDamageChance, output.LightningSpellProjectileDamageChance, output.ColdSpellProjectileDamageChance, output.FireSpellProjectileDamageChance, output.ChaosSpellProjectileDamageChance})
+	
+	--maximum hit taken
+	--FIX X TAKEN AS Y (output[damageType.."TotalPool"] should use the damage types that are converted to in output[damageType.."TakenHitMult"])
+	for _, damageType in ipairs(dmgTypeList) do
+		output[damageType.."MaximumHitTaken"] = output[damageType.."TotalPool"] / output[damageType.."TakenHitMult"]
+		if breakdown then
+			breakdown[damageType.."MaximumHitTaken"] = {
+				s_format("Total Pool: %d", output[damageType.."TotalPool"]),
+				s_format("Damage Taken modifier: %.2f", output[damageType.."TakenHitMult"]),
+				s_format("Maximum hit you can take: %d", output[damageType.."MaximumHitTaken"]),
+			}
+		end
+	end
+	output.SecondMinimumMaximumHitTaken = m_second_min({output.PhysicalMaximumHitTaken, output.LightningMaximumHitTaken, output.ColdMaximumHitTaken, output.FireMaximumHitTaken, output.ChaoSecondMaximumimumHitTaken})
+	
+	--effective health pool vs dots
+	for _, damageType in ipairs(dmgTypeList) do
+		output[damageType.."DotEHP"] = output[damageType.."TotalPool"] / output[damageType.."TakenDotMult"]
+		if breakdown then
+			breakdown[damageType.."DotEHP"] = {
+				s_format("Total Pool: %d", output[damageType.."TotalPool"]),
+				s_format("Dot Damage Taken modifier: %.2f", output[damageType.."TakenDotMult"]),
+				s_format("Total Effective Dot Pool: %d", output[damageType.."DotEHP"]),
+			}
+		end
+	end
+	output.SecondMinimumDotEHP = m_second_min({output.PhysicalDotEHP, output.LightningDotEHP, output.ColdDotEHP, output.FireDotEHP, output.ChaosDotEHP})
+	
+	--total EHP
+	for _, damageType in ipairs(dmgTypeList) do
+		local convertedAvoidance = 0
+		for _, damageConvertedType in ipairs(dmgTypeList) do
+			convertedAvoidance = convertedAvoidance + output[damageConvertedType.."AverageDamageChance"] * actor.damageShiftTable[damageType][damageConvertedType] / 100
+		end
+		output[damageType.."TotalEHP"] = output[damageType.."MaximumHitTaken"] / (1 - output.AverageNotHitChance / 100) / (1 - convertedAvoidance / 100)
+		if breakdown then
+			breakdown[damageType.."TotalEHP"] = {
+			s_format("Maximum Hit taken: %d", output[damageType.."MaximumHitTaken"]),
+			s_format("Average chance not to be hit: %d%%", output.AverageNotHitChance),
+			s_format("Average chance to not take damage when hit: %d%%", convertedAvoidance),
+			s_format("Total Effective Hit Pool: %d", output[damageType.."TotalEHP"]),
+			}
+		end
+	end
+	output.SecondMinimumTotalEHP = m_second_min({output.PhysicalTotalEHP, output.LightningTotalEHP, output.ColdTotalEHP, output.FireTotalEHP, output.ChaosTotalEHP})
```

Comment:
Also wondering if we could condense this into a loop... potentially not possible.

Reply:
It would be possible to do a minor nested loop, as there are 2 uses of block chance and 2 of spell chance, so you can do a loop like
for _, X in ipairs({"Attack", "Spell"}) do
but it would barely change the size, make it less readable and require the renaming of multiple variables